bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,269 @@
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand(channelUID:{}, command:{})", channelUID.getAsString(), command.toFullString());
+
+        if (command instanceof RefreshType) {
+            refreshPrices();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        if (historicDaysInitially == 0) {
+            historicDaysInitially = config.historicDays;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+        refreshJob = scheduler.schedule(this::refreshPrices, 0, TimeUnit.SECONDS);
+    }
+
+    private ZonedDateTime currentCetTime() {
+        // Use region-based ZoneId instead of CET to handle transition to CEST automatically
+        return ZonedDateTime.now(ZoneId.of("Europe/Paris"));
+    }
+
+    private ZonedDateTime currentCetTimeWholeHours() {
+        return currentCetTime().truncatedTo(ChronoUnit.HOURS);
+    }
+
+    private BigDecimal getCurrentHourKwhPrice() throws EntsoeResponseMapException {
+        if (responseMap.isEmpty()) {
+            throw new EntsoeResponseMapException("responseMap is empty");
+        }
+        var currentCetTime = currentCetTime();
+        Integer currentHour = currentCetTime.getHour();
+        Integer currentDayInYear = currentCetTime.getDayOfYear();
+        var result = responseMap.entrySet().stream()
+                .filter(x -> x.getKey().getHour() == currentHour && x.getKey().getDayOfYear() == currentDayInYear)
+                .findFirst();
+        if (result.isEmpty()) {
+            throw new EntsoeResponseMapException(
+                    String.format("Could not find a value for hour {} and day in year {} in responseMap", currentHour,
+                            currentDayInYear));
+        }
+
+        double eurMwhPrice = result.get().getValue();
+        BigDecimal kwhPrice = BigDecimal.valueOf(eurMwhPrice).divide(new BigDecimal(1000), 10, RoundingMode.HALF_UP);
+        return kwhPrice;
+    }
+
+    private boolean needToFetchHistoricDays() {
+        return needToFetchHistoricDays(false);
+    }
+
+    private boolean needToFetchHistoricDays(boolean updateHistoricDaysInitially) {
+        boolean needToFetch = false;
+        if (historicDaysInitially < config.historicDays) {
+            logger.debug("Need to fetch historic data. Historicdays was changed to a greater number: {}",
+                    config.historicDays);
+            needToFetch = true;
+        }
+
+        if (updateHistoricDaysInitially && historicDaysInitially != config.historicDays) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        return needToFetch;
+    }
+
+    private void refreshPrices() {
+        if (!isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
+            logger.debug("Channel {} is not linked, cant update channel", EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
```

Comment:
Actually we can, we just don't need to. It's an optimization.
```suggestion
            logger.debug("Channel {} is not linked, skipping channel update", EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
```